src/libostree/ostree-repo-traverse.c

```diff
@@ -660,6 +680,33 @@ ostree_repo_traverse_commit_union (OstreeRepo      *repo,
                                                     cancellable, error);
 }
 
+/**
+ * ostree_repo_traverse_commit_only_union: (skip)
+ * @repo: Repo
+ * @commit_checksum: ASCII SHA256 checksum
+ * @maxdepth: Traverse this many parent commits, -1 for unlimited
+ * @inout_reachable: Set of reachable objects
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Update the set @inout_reachable containing all COMMIT objects reachable
+ * from @commit_checksum, traversing @maxdepth parent commits.
+ */
+gboolean
+ostree_repo_traverse_commit_only_union (OstreeRepo      *repo,
```

Comment:
I think what we should do instead is have
`ostree_repo_traverse_commit_with_flags` that accepts `OstreeRepoCommitTraverseFlags` as a parameter too.

Another way to say this is just to make `ostree_repo_traverse_commit_internal()` the public API.

To restate another way, this is I think fixing a historical bug that this API didn't take flags; it's only the explicit external iteration API (`ostree_repo_commit_traverse_iter_init_commit`) that takes flags right now.

Reply:
I agree, that sounds like a good idea to me. When I was thinking about this, I was worried about doubling the functionality of `ostree_repo_traverse_commit_union` and potentially making that function obsolete. Is that a concern here?

Reply:
We are indeed making that function obsolete, but that's fine!  I don't think we need to mark it as explicitly deprecated or anything.

See also https://lwn.net/Articles/585415/